mslib/mscolab/seed.py

```diff
@@ -226,178 +226,175 @@ def archive_operation(path=None, emailid=None):
-            'description': 'b, c',
-            'category': 'default'
-        }, {
-            'id': 3,
-            'path': 'three',
-            'description': 'a, c',
-            'category': 'default'
-        }, {
-            'id': 4,
-            'path': 'four',
-            'description': 'd',
-            'category': 'default'
-        }, {
-            'id': 5,
-            'path': 'Admin_Test',
-            'description': 'Operation for testing admin window',
-            'category': 'default'
-        }, {
-            'id': 6,
-            'path': 'test_mscolab',
-            'description': 'Operation for testing mscolab main window',
-            'category': 'default'
-        }]
-        for operation in operations:
-            db_operation = Operation(operation['path'], operation['description'], operation['category'])
-            db_operation.id = operation['id']
-            db.session.add(db_operation)
-
-        # create permissions
-        permissions = [{
-            'u_id': 8,
-            'op_id': 1,
-            'access_level': "creator"
-        }, {
-            'u_id': 9,
-            'op_id': 1,
-            'access_level': "collaborator"
-        }, {
-            'u_id': 9,
-            'op_id': 2,
-            'access_level': "creator"
-        }, {
-            'u_id': 10,
-            'op_id': 2,
-            'access_level': "collaborator"
-        }, {
-            'u_id': 10,
-            'op_id': 3,
-            'access_level': "creator"
-        }, {
-            'u_id': 8,
-            'op_id': 3,
-            'access_level': "collaborator"
-        }, {
-            'u_id': 10,
-            'op_id': 1,
-            'access_level': "viewer"
-        }, {
-            'u_id': 11,
-            'op_id': 4,
-            'access_level': 'creator'
-        }, {
-            'u_id': 8,
-            'op_id': 4,
-            'access_level': 'admin'
-        }, {
-            'u_id': 13,
-            'op_id': 3,
-            'access_level': 'viewer'
-        }, {
-            'u_id': 12,
-            'op_id': 5,
-            'access_level': 'creator'
-        }, {
-            'u_id': 12,
-            'op_id': 3,
-            'access_level': 'collaborator'
-        }, {
-            'u_id': 15,
-            'op_id': 5,
-            'access_level': 'viewer'
-        }, {
-            'u_id': 14,
-            'op_id': 3,
-            'access_level': 'collaborator'
-        }, {
-            'u_id': 15,
-            'op_id': 3,
-            'access_level': 'collaborator'
-        }, {
-            'u_id': 16,
-            'op_id': 6,
-            'access_level': 'creator'
-        }, {
-            'u_id': 17,
-            'op_id': 6,
-            'access_level': 'admin'
-        }]
-        for perm in permissions:
-            db_perm = Permission(perm['u_id'], perm['op_id'], perm['access_level'])
-            db.session.add(db_perm)
-        db.session.commit()
-        db.session.close()
+    # create users
```

Comment:
observation:

using seed now downgrades first
```

(mssdev) reimar@infinity:~/PycharmProjects/2023/matrss/MSS$ python mslib/mscolab/mscolab.py db --seed
WARNING:root:Couldn't import mscolab_settings (ImportError:'No module named 'mscolab_settings''), using dummy config.
WARNING:root:Couldn't import setup_saml2_backend (ImportError:'No module named 'setup_saml2_backend''), using dummy config.
INFO  [alembic.runtime.migration] Context impl SQLiteImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
Database initialised successfully!
WARNI [root] Couldn't import mscolab_auth (ImportError:'{No module named 'mscolab_auth'), creating dummy config.
Are you sure you want to seed the database? Seeding will delete all your existing data and replace it with seed data (y/[n]):y
INFO  [alembic.runtime.migration] Context impl SQLiteImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
INFO  [alembic.runtime.migration] Running downgrade 922e4d9c94e2 -> c171019fe3ee, To version 10.0.0
INFO  [alembic.runtime.migration] Running downgrade c171019fe3ee -> 92eaba86a92e, To version 9.0.0
INFO  [alembic.runtime.migration] Running downgrade 92eaba86a92e -> , To version 8.3.5 - Initial migration
INFO  [alembic.runtime.migration] Context impl SQLiteImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
INFO  [alembic.runtime.migration] Running upgrade  -> 92eaba86a92e, To version 8.3.5 - Initial migration
INFO  [alembic.runtime.migration] Running upgrade 92eaba86a92e -> c171019fe3ee, To version 9.0.0
INFO  [alembic.runtime.migration] Running upgrade c171019fe3ee -> 922e4d9c94e2, To version 10.0.0
Database seeded successfully!

```

Reply:
What it does now is a downgrade to the base revision (i.e. a empty database) and then upgrade again. This drops all tables and recreates them, removing all content along the way.

This is the same operation that reset already did (drop_all followed by create_all), just rephrased in terms of flask_migrate. I don't really see a reason to change that, unless it ever becomes unbearably slow due to too many migrations.

Doing it this way is even a must, if we understand reset not just in terms of the data in the database, but also in terms of the schema of the database. Imagine someone (accidentally?) modifies the schema, e.g. adds a column or an index, to the database. Now, if reset didn't do a downgrade + upgrade but just deleted all rows it wouldn't be able to reset the schema back to what it should be.